crates/rpc-client/src/client.rs

```diff
@@ -118,15 +146,44 @@ pub struct RpcClientInner<T> {
 
 impl<T> RpcClientInner<T> {
     /// Create a new [`RpcClient`] with the given transport.
+    #[inline]
     pub const fn new(t: T, is_local: bool) -> Self {
         Self { transport: t, is_local, id: AtomicU64::new(0) }
     }
 
+    /// Returns the default poll interval for the client.
+    pub fn default_poll_interval(&self) -> Duration {
```

Comment:
we could also get a blocktime hint here by querying 2 blocks and subtracting their timestamps

Reply:
Would have to be set from the heartbeat task and the field would have to be a `Arc<RwLock<Duration>>`, which means we can also synchronize all polling intervals globally for that provider? I can see this happening